x-pack/platform/plugins/shared/cases/public/client/helpers/get_observables_from_ecs.ts

```diff
@@ -0,0 +1,162 @@
+  }
+  return 'IPV4';
+};
+
+// https://www.elastic.co/docs/reference/ecs/ecs-hash
+const HASH_FIELDS = [
+  'cdhash',
+  'md5',
+  'sha1',
+  'sha256',
+  'sha384',
+  'sha512',
+  'ssdeep',
+  'tlsh',
+] as const;
+
+// https://www.elastic.co/docs/reference/ecs/ecs-hash
+// TODO - support 'email.attachments.file'
+// https://github.com/elastic/security-team/issues/13709
+const HASH_PARENTS = ['dll', 'file', 'process'] as const;
+
+export const getHashValues = (ecsData: Ecs): string[] => {
+  const res: string[] = [];
+
+  HASH_PARENTS.forEach((parent) => {
+    HASH_FIELDS.forEach((field) => {
+      const value = ecsData[parent]?.hash?.[field];
+      if (value && typeof value === 'string') {
+        res.push(value);
+      } else if (Array.isArray(value)) {
+        res.push(...value);
+      }
+    });
+  });
+
+  return res;
+};
+
+export const getObservablesFromEcs = (ecsData: Ecs): ObservablePost[] => {
+  const observables = [];
+  const description = i18n.translate(
+    'xpack.cases.caseView.observables.autoExtractedObservablesDescription',
+    {
+      defaultMessage: 'Auto extracted observable',
+    }
+  );
+
+  // Source IP
+  if (ecsData.source?.ip) {
+    const ips = Array.isArray(ecsData.source?.ip) ? ecsData.source?.ip : [ecsData.source?.ip];
+    ips.forEach((ip) => {
+      const ipType = getIPType(ip);
+      observables.push({
+        typeKey: ipType === 'IPV4' ? 'observable-type-ipv4' : 'observable-type-ipv6',
+        value: ip,
+        description,
+      });
+    });
+  }
+
+  // Destination IP
+  if (ecsData.destination?.ip) {
+    const ips = Array.isArray(ecsData.destination.ip)
+      ? ecsData.destination.ip
+      : [ecsData.destination.ip];
+
+    ips.forEach((ip) => {
+      const ipType = getIPType(ip);
+      observables.push({
+        typeKey: ipType === 'IPV4' ? 'observable-type-ipv4' : 'observable-type-ipv6',
+        value: ip,
+        description,
+      });
+    });
+  }
+
+  // Host name
+  if (ecsData.host?.name) {
+    const hostnames = Array.isArray(ecsData.host?.name) ? ecsData.host?.name : [ecsData.host?.name];
+    hostnames.forEach((name) => {
+      observables.push({
+        typeKey: 'observable-type-hostname',
+        value: name,
+        description,
+      });
+    });
+  }
+
+  // File hash
+  const hashValues = getHashValues(ecsData);
+  if (hashValues.length > 0) {
+    observables.push(
+      ...hashValues.map((hash) => ({
+        typeKey: 'observable-type-file-hash',
+        value: hash,
+        description,
+      }))
+    );
+  }
+
+  // File path
+  if (ecsData.file?.path) {
+    const paths = Array.isArray(ecsData.file?.path) ? ecsData.file?.path : [ecsData.file?.path];
+    paths.forEach((path) => {
+      observables.push({
+        typeKey: 'observable-type-file-path',
+        value: path,
+        description,
+      });
+    });
+  }
+
+  // Domain;
+  if (ecsData.dns?.question?.name) {
+    const names = Array.isArray(ecsData.dns?.question?.name)
```

Comment:
Overthinking but, you could create a util if it doesn't existing in the codebase `const getValueAsArray = (val: NonNullable<unknown>) => (Array.isArray(val) ? val : [val]);` and reuse it in this file

Reply:
lodash has `castArray`:  `const names = castArray(ecsData.dns?.question?.name ?? []);` (sorry for the driveby)